pyccl/halos/halo_model.py

```diff
@@ -79,6 +83,14 @@ def __init__(self, *, mass_function, halo_bias, mass_def=None,
         self._cosmo_mf = self._cosmo_bf = None
         self._a_mf = self._a_bf = -1
 
+    def _fix_profile_mass_def(self, prof):
+        # TODO v3: remove this (in v3 all profiles have a mass_def).
+        # If profile has no mass definition assigned, assign one.
+        if prof.mass_def is None:
+            warnings.warn("In v3 all profiles will need an associated "
+                          "mass definition.", CCLDeprecationWarning)
+            prof.mass_def = self.mass_def
+
```

Comment:
This was necessary in order to ensure that we could use profiles that were instantiated without a mass definition attached to them from scratch (this will not be possible in v3, but it was in v2).